File: 1030-matrix-cells-in-distance-order/1030-matrix-cells-in-distance-order.py
class Solution:
    def allCellsDistOrder(self, rows: int, cols: int, rCenter: int, cCenter: int) -> List[List[int]]:
        
        def my_sort(coord: List[int]):
            r, c = coord
            return abs(r - rCenter) + abs(c - cCenter)
        
        arr = [[r, c] for c in range(cols) for r in range(rows)]
        return sorted(arr, key=my_sort)
            

File: 1030-matrix-cells-in-distance-order/1030-matrix-cells-in-distance-order.js
/**
 * @param {number} rows
 * @param {number} cols
 * @param {number} rCenter
 * @param {number} cCenter
 * @return {number[][]}
 */
var allCellsDistOrder = function(rows, cols, rCenter, cCenter) {
    const res = []
    for (let r=0; r < rows; r++) {
        for (let c=0; c < cols; c++) {
            res.push([r, c])
        }
    }
    return res.sort(mySort)
    
    function mySort([a, b], [c, d]) {
        return (Math.abs(a - rCenter) + Math.abs(b - cCenter)) < (Math.abs(c - rCenter) + Math.abs(d - cCenter)) ? -1 : 1
    }
};